infra/modules/azure_function_app/locals.tf

```diff
@@ -39,11 +34,11 @@ locals {
 
   storage_account = {
     replication_type = local.tier == "s" ? "LRS" : "ZRS"
-    name             = replace("${local.project}${replace(local.domain, "-", "")}${var.environment.app_name}stfn${var.environment.instance_number}", "-", "")
-    pep_blob_name    = "${local.app_name_prefix}-blob-pep-${var.environment.instance_number}"
-    pep_file_name    = "${local.app_name_prefix}-file-pep-${var.environment.instance_number}"
-    pep_queue_name   = "${local.app_name_prefix}-queue-pep-${var.environment.instance_number}"
-    alert            = "[${replace("${local.project}${replace(local.domain, "-", "")}${var.environment.app_name}stfn${var.environment.instance_number}", "-", "")}] Low Availability"
+    name             = replace("${module.naming_convention.project}${replace(module.naming_convention.domain, "-", "")}${var.environment.app_name}stfn${module.naming_convention.suffix}", "-", "")
```

Comment:
what about lowercasing and limiting to 24 chars? something like:

```
  name = substr( replace(lower("${module.naming_convention.prefix}st${module.naming_convention.suffix}"), "/[^a-z0-9]/", ""),
    0,
    24
  )

```

Reply:
Regarding the 24 character limit, it is already done during the validation phase of the environment variable, and it is better to leave it like that, doing checks in locals can lead to drift in the state in case of errors during apply. 

The `lower(...)` can be done, it must first be checked if a function with capital letters has ever been generated, because otherwise backward compatibility would be lost